tls/s2n_async_pkey.c

```diff
@@ -641,3 +641,77 @@ S2N_RESULT s2n_async_pkey_op_copy_hash_state_for_testing(struct s2n_async_pkey_o
     RESULT_GUARD_POSIX(s2n_hash_copy(copy, &op->op.sign.digest));
     return S2N_RESULT_OK;
 }
+
+DEFINE_POINTER_CLEANUP_FUNC(struct s2n_hash_state *, s2n_hash_free);
+
+DEFINE_POINTER_CLEANUP_FUNC(struct s2n_blob *, s2n_free);
+
+static S2N_RESULT s2n_async_pkey_verify_perform(struct s2n_async_op *op)
+{
+    RESULT_ENSURE_REF(op);
+    RESULT_ENSURE_REF(op->conn);
+    RESULT_ENSURE(op->type == S2N_ASYNC_PKEY_VERIFY, S2N_ERR_INVALID_ARGUMENT);
+
```

Comment:
S2N_ERR_INVALID_ARGUMENT is a usage error, indicating that the customer did something to trigger it. Shouldn't this be an internal error? Could you end up with the wrong type without a bug in s2n-tls?